src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/actions/GenerateTestsActionLine.kt

```diff
@@ -0,0 +1,54 @@
+package nl.tudelft.ewi.se.ciselab.testgenie.actions
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.Caret
+import com.intellij.psi.PsiFile
+import nl.tudelft.ewi.se.ciselab.testgenie.evosuite.Runner
+
+/**
+ * This class contains all the logic related to generating tests for a line.
+ * No actual generation happens in this class, rather it is responsible for displaying the action option to the user when it is available,
+ *   getting the information about the selected class and passing it to (EvoSuite) Runner.
+ */
+class GenerateTestsActionLine : AnAction() {
+    /**
+     * Creates and calls (EvoSuite) Runner to generate tests for a line when the action is invoked.
+     *
+     * @param e an action event that contains useful information and corresponds to the action invoked by the user
+     */
+    override fun actionPerformed(e: AnActionEvent) {
+        val evoSuiteRunner: Runner = createEvoSuiteRunner(e) ?: return
+
+        val psiFile: PsiFile = e.dataContext.getData(CommonDataKeys.PSI_FILE) ?: return
+        val caret: Caret = e.dataContext.getData(CommonDataKeys.CARET)?.caretModel?.primaryCaret ?: return
+
+        val selectedLine: Int = getSurroundingLine(psiFile, caret)?.plus(1)
+            ?: return // lines in the editor and in EvoSuite are one-based
+
+        Logger.getInstance("GenerateTestsUtils").info("Selected line is $selectedLine")
```

Comment:
You should use a logger corresponding to this class (e.g. `Logger.getInstance(this.javaClass)`)

Reply:
The problem is that GenerateTestsUtils is not a class, hence i cannot do this trick with it. So, for consistency, i named all of them as "GenerateTestsUtils"

Reply:
But this is in the GenerateTestsActionLine class, which is a class. You can do this here.

You shouldn't instantiate loggers named after a different class since that creates confusion.

Reply:
How do i name the one for GenerateTestsUtils?

Reply:
Done